packages/flutter_tools/lib/src/dart/analysis.dart

```diff
@@ -125,18 +122,18 @@ class AnalysisServer {
         final String event = response['event'] as String;
         final dynamic params = response['params'];
 
-        if (params is Map<String, dynamic>) {
+        if (params is Map<String, dynamic> && castStringKeyedMap(response['params']) != null) {
```

Comment:
Would it help clean this up a little to pull `castStringKeyedMap(response['params'])` out into a local?

Reply:
Done.